[ENH] `SignatureTransformer` and `SignatureClassifier` `numpy 2` compatibility

`SignatureTransformer` and `SignatureClassifier` fail under `numpy 2`, this should be upgraded.

The failure is

```
>   from . import _tosig as tosig
E   ImportError: numpy.core.multiarray failed to import
```

--- a/sktime/transformations/panel/signature_based/_signature_method.py
+++ b/sktime/transformations/panel/signature_based/_signature_method.py
@@ -61,7 +61,7 @@ class SignatureTransformer(BaseTransformer):
         # --------------
         "authors": "jambo6",
         "maintainers": "jambo6",
-        "python_dependencies": "esig",
+        "python_dependencies": ["esig", "numpy<2.0"],
         "python_version": "<3.10",
         # estimator type
         # --------------

